### PR TITLE
adding revised learning objectives

### PR DESCRIPTION
**Pull request by cmilica**

### CODE REVIEW

File: learning_objectives.md [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
-|Be prepared to explore opportunities of applying ML in your own projects.**(help needed here)***  |    |
-| understanding specific models? Which models?   |   |
-| understanding pipeline need to figure out how to word this  |   |
-| unique issues in biological applications?    |   |
+| Identify the characteristics of machine learning and describe the ML fundamentals.| Define machine learning and conceptualize the core principles and classification. |
+| Read papers that apply ML to biological problems with comprehension.| Identify current problems in computational biology and recognize the benefits from machine learning techniques and the modern big data technologies.|
+| Comprehend how the datasets are preprocesses, what are samples, features and class labels, what is model in machine learning, what is training set, and hyperparameters, validation set, evaluation and prediction, and how do we perform model selection. | Explain and recognize the language used by computational biologists and be more comfortable to collaborate with them on the projects that use machine learning.|
```
Comment (agitter) [inline learning_objectives.md]: `preprocesses` -> `preprocessed`

We could shorten this outcome by saying something about dataset characteristics and ML workflow steps instead of listing each term here.

File: learning_objectives.md [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
+| Read papers that apply ML to biological problems and judge whether ML has been used appropriately and effectively. |  Compare and contrast strengths and weaknesses of many machine learning classifiers commonly used in biology and differentiate whether the process of looking at classifiers trained with different hypermarameters led to a bad model that fails on new data.|
```
Comment (agitter) [inline learning_objectives.md]: These may be two separate ideas.  One has to do with reading papers or understanding a ML workflow that was performed and assessing whether it followed best practices.  The other is about strengths and weaknesses of individual classifiers, which may fit better with the specific models below.

File: learning_objectives.md [hunk 3]
```diff
@@ -2,13 +2,14 @@
...
+| Assess model selection and recognize that these methods don't necessarily work right out of the box. |  Evaluate ML models and understand their complexity and their limitations. Even simple models have multiple versions (hyperparameters) that needs to be tuned for those hyperparameters and the best version of a model needs to be chosed that will make accurate predictions on new data. |
+|Select and evaluate a model on a case-study dataset. | Demostrate understanding how to use a dataset to select and evaluate a model following the Ml workflow (cross-valudation or traing, validation  and test splitting).   |
```
Comment (agitter) [inline learning_objectives.md]: `valudation` -> `validation`
`traing` -> `training`

File: learning_objectives.md [hunk 4]
```diff
@@ -2,13 +2,14 @@
...
+| Apply the model to new data. | Discuss what to do once the model has been trained and what to watch out for in appkying to future data.  |
```
Comment (agitter) [inline learning_objectives.md]: `appkying` -> `applying`

File: learning_objectives.md [hunk 5]
```diff
@@ -2,13 +2,14 @@
...
+| specific models - logistic regression, decision trees, neural networks   |  |
+| Predicting cell, tissue, or organism properties from "omic" data (issues of sample size versus number of features), imaging (issues likely related to structured features and suitability for models like convolutional neural networks, which is out of our scope), genome annotation (issues related to training and evaluation on skewed datasets dominated by negative instances).  | Identify and investigate the unique issues in biological applications.  |
```
Comment (agitter) [inline learning_objectives.md]: Does this outcome fit with the previous `Identify current problems in computational biology and recognize the benefits from machine learning techniques and the modern big data technologies.`?  I see these as examples of current problems in computational biology.

File: learning_objectives.md (hunk 2 above)
Comment (agitter) [reply learning_objectives.md]: In addition, do we want to combine the two lines related to "reading papers"?  Or are those distinct enough goals to remain separate?

File: learning_objectives.md [hunk 6]
```diff
@@ -2,13 +2,14 @@
...
+| Comprehend how the datasets are preprocesses, what are samples, features and class labels, what is model in machine learning, what is training set, and hyperparameters, validation set, evaluation and prediction, and how do we perform model selection. | Explain and recognize the language used by computational biologists and be more comfortable to collaborate with them on the projects that use machine learning.|
+| Read papers that apply ML to biological problems and judge whether ML has been used appropriately and effectively. |  Compare and contrast strengths and weaknesses of many machine learning classifiers commonly used in biology and differentiate whether the process of looking at classifiers trained with different hypermarameters led to a bad model that fails on new data.|
+| Assess model selection and recognize that these methods don't necessarily work right out of the box. |  Evaluate ML models and understand their complexity and their limitations. Even simple models have multiple versions (hyperparameters) that needs to be tuned for those hyperparameters and the best version of a model needs to be chosed that will make accurate predictions on new data. |
```
Comment (agitter) [inline learning_objectives.md]: I like this idea a lot.  We should think about how to make the phrasing more compact.

File: learning_objectives.md [hunk 7]
```diff
@@ -2,13 +2,14 @@
...
+|Select and evaluate a model on a case-study dataset. | Demostrate understanding how to use a dataset to select and evaluate a model following the Ml workflow (cross-valudation or traing, validation  and test splitting).   |
+| Apply the model to new data. | Discuss what to do once the model has been trained and what to watch out for in appkying to future data.  |
+| specific models - logistic regression, decision trees, neural networks   |  |
```
Comment (agitter) [inline learning_objectives.md]: Let's complete this thought.  Is the goal to understand how these models work so they aren't black boxes?  And perhaps also to understand differences between them, most notably between linear models and non-linear models?  As I mentioned above, that relates to `Compare and contrast strengths and weaknesses...`.

File: learning_objectives.md (hunk 5 above)
Comment (agitter) [reply learning_objectives.md]: We may not need to list these as a separate example.  I'd argue the main objective is to recognize applications, challenges, and opportunities specific to biology.

Comment (cmilica) [reply learning_objectives.md]: This was all of my thoughts just poured out in one place before we finalize it. I do like "Recognize applications, challenges, and opportunities specific to biology." as an objective. I was wondering what would be the way that we address it in the workshop?

File: learning_objectives.md (hunk 6 above)
Comment (cmilica) [reply learning_objectives.md]: For line 8 - We should combine the "reading papers" lines. Or rephrase the second learning outcome. 

For line 9 - maybe once we know exactly how we are going to address this outcome it would be easier to phrase it better?

File: learning_objectives.md (hunk 7 above)
Comment (cmilica) [reply learning_objectives.md]: I was thinking we should be explicit in stating the outcome. So that as the outcome of the workshop participants will be able to: "apply specific classifiers like logistic regression, decision trees, and neural networks and make sense of the prediction outcomes". This is very broad, but maybe could help explain the direction I was going in with this Objective.

File: learning_objectives.md (hunk 6 above)
Comment (agitter) [reply learning_objectives.md]: Feedback from discussion with Beth

> For this one, perhaps we could simplify and focus on the point that every type of model has multiple versions controlled by hyperparameters.  The modeler has to make decisions about which version to use.  This is something we plan to show in the workshop early on by considering a one-layer decision tree that splits on a single feature versus a two-layer decision tree that has two splits.

File: learning_objectives.md (hunk 4 above)
Comment (agitter) [reply learning_objectives.md]: Feedback from Discussion with Beth

> I see this as an important problem for machine learning in biomedicine.  It is hard to know when assumptions made when training a model, for instance about data being independent and identically distributed, no longer hold on a new dataset to which you want to apply the model.  Maybe that is too advanced for this audience.  Should we recommend they consult with experts when considering this?  Or not mention it at all?

File: learning_objectives.md (hunk 3 above)
Comment (agitter) [reply learning_objectives.md]: Feedback from discussion with Beth

> Part of this was related to the idea above.  If we have two versions of a model (e.g. a decision tree) we can apply it to validation set data it was not trained on to estimate generalization performance.  Would this be more realistic if we narrow it to evaluating a model with a single validation set?  We could drop cross validation for this beginners group.
